lib/src/view/study/study_screen.dart

```diff
@@ -11,6 +40,460 @@ class StudyScreen extends ConsumerWidget {
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
-    return const SizedBox.shrink();
+    final state = ref.watch(studyControllerProvider(id));
+
+    return state.when(
+      data: (state) {
+        return PlatformScaffold(
+          appBar: PlatformAppBar(
+            title: FittedBox(
+              fit: BoxFit.scaleDown,
+              child: Text(state.currentChapterTitle),
+            ),
+            actions: [
+              AppBarIconButton(
+                onPressed: () => showAdaptiveBottomSheet<void>(
+                  context: context,
+                  isScrollControlled: true,
+                  showDragHandle: true,
+                  isDismissible: true,
+                  builder: (_) => StudySettings(id),
+                ),
+                semanticsLabel: context.l10n.settingsSettings,
+                icon: const Icon(Icons.settings),
+              ),
+              _ChapterButton(id: id),
+            ],
+          ),
+          body: _Body(id: id),
+        );
+      },
+      loading: () {
+        return const PlatformScaffold(
+          appBar: PlatformAppBar(
+            title: Text(''),
+          ),
+          body: Center(child: CircularProgressIndicator()),
+        );
+      },
+      error: (error, st) {
+        _logger.severe('Cannot load study: $error', st);
+        Navigator.of(context).pop();
```

Comment:
I never used that pattern in the rest of the app. Not sure it is good UX wise. But for consistency's sake we should display an error instead and let the user come back.